### ISSUE TITLE
Testcase 567: missing LIB="" in state-env.c-native-r2

### ISSUE THREAD

**Issue by jshimoda9323** [ZOSOpenTools/autoconfport#17]
I've been trying to investigate failing testcase 567.  So far here's what I've found:
tests/testsuite.dir/567/testsuite.log contains this at the end:
```configure: creating ./config.status
config.status: creating config.h
config.status: config.h is unchanged
./acfunctions.at:10: ath_fn_check_env
./acfunctions.at:10: $at_diff "config-h.c-native-r1" "config-h.c-native-r2"
./acfunctions.at:10: $at_diff "at_config_vars-state-env.c-native-r1" "at_config_vars-state-env.c-native-r2"
0a1,2
> 34d33
> < LIBS=""
./acfunctions.at:10: exit code was 1, expected 0
```
It seems that the test framework is diffing the files `state-env.c-native-r1` and `state-env.c-native-r2` which are produced during a run of `configure -C`, and then a subsequent run of `configure -C` respectively.  The second run picks up `config.cache` file produced by the first run it seems.
File `state-env.c-native-r1` contains an extra `LIBS=""` that `state-env.c-native-r2` does not.  So far I do not know why.

**Comment by jshimoda9323** [comment]
Mike suggested that I try setting `LIBS=""` before running the test and it worked - the test was successful....
It seems that configure is generated in such a way that unconditionally sets LIBS="" in it's cache file, even if it wasn't set when configure originally ran.  This causes the test framework to trigger a failure when comparing the environment variables between the first run of configure (LIBS is unset), and the second when it picks up the cache (LIBS="", which is the zero length string).  So anyway....there's two possible fixes, I think.  1. FIx autoconf to generate configure such that it only sets LIBS in the cache file if LIBS has a value originally.  or 2. Modify the test framework to ignore LIBS in this case where LIBS="".

**Comment by jshimoda9323** [comment]
Investigated further.  The problem is actually that LIBS is set during the first run of `configure`, so it appears in state-env.c-native.r1, but on the second run of configure when the cache is picked up, LIBS is not set, so it does not appear in state-env.c-native.r2.  (I was confused in my previous comment).  There is a macro that is generating script that sets LIBS in the first case, but not the second.  So then I did a search on this problem to see if anyone else ran into it before....
Turns out yes, someone in 2005 ran into a related problem and there is a fix.
https://lists.gnu.org/archive/html/bug-autoconf/2005-07/msg00018.html
I'm proposing a simliar fix - I sent an email off to autoconf-patches to get this upstreamed.
